osu.Game/Users/UserPanel.cs

```diff
@@ -179,9 +192,34 @@ public MenuItem[] ContextMenuItems
 
                 bool isUserOnline() => metadataClient?.GetPresence(User.OnlineID) != null;
                 bool canInviteUser() => isUserOnline() && multiplayerClient?.Room?.Users.All(u => u.UserID != User.Id) == true;
+                bool isUserBlocked() => api.Blocks.Any(b => b.TargetID == User.OnlineID);
             }
         }
 
+        private void blockUser(bool block)
+        {
+            loading.Show();
+            APIRequest req = block ? new BlockUserRequest(User.OnlineID) : new UnblockUserRequest(User.OnlineID);
+
+            req.Success += () =>
+            {
+                api.UpdateLocalBlocks();
```

Comment:
This is super awkward. I've much rather the `Block` / `Unblock` methods are on a global class.

But... it does match the implementation of `UpdateLocalFriends` so I can't really fault the implementation.